### PR TITLE
Gitlab CI for careers

### PR DESCRIPTION
**Pull request by metadave**
headers set via Lambda@Edge:
https://careers.allizom.org/

todo
- [ ] remove netlify integrations
- [x] ~implement lambda@edge stuff~
  - https://github.com/mozmeao/infra/pull/1052

### CODE REVIEW

**Comment by metadave** [comment]
@pmac @glogiotatidis any idea on how we disable the netlify checks above?

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1,11 +1,16 @@
-netlify:
-  type: deploy
+image: python:3-slim-stretch
```
Comment (pmac) [inline .gitlab-ci.yml]: Are we using shared runners for this instead of our own?

Comment (metadave) [reply .gitlab-ci.yml]: I disabled shared runners for this project, so it's using our own.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
-    - curl -X POST -d {} ${WEBHOOK_PROD}
-    # Dead Man's Snitch Ping
-    - curl ${DMS}
-  only:
-    - schedules
+     - pip install awscli
+     - pip install --require-hashes --no-cache-dir -r requirements.txt    
+     - ./bin/mirror.sh
+     - aws s3 sync ./_site ${STAGE_BUCKET_PATH} --acl public-read --delete
+     - aws s3 sync ./_site ${PROD_BUCKET_PATH} --acl public-read --delete
```
Comment (pmac) [inline .gitlab-ci.yml]: Why have stage if they're going to be exactly the same?

Comment (metadave) [reply .gitlab-ci.yml]: copied from [here](https://github.com/mozmeao/lumbergh/blob/master/.gitlab-ci.yml#L5). I'm not sure. any ideas @glogiotatidis ?

File: .gitlab-ci.yml (hunk 1 above)
Comment (pmac) [reply .gitlab-ci.yml]: then we don't need `image` and it shouldn't be running any `apt-get` things for every run.

File: .gitlab-ci.yml (hunk 2 above)
Comment (pmac) [reply .gitlab-ci.yml]: oh. that's because netlify was building those two deployments from different branches.

Comment (pmac) [reply .gitlab-ci.yml]: while that job was set to run only on a schedule and update those deployments from those branches. To do the same thing here we'll need separate jobs for stage and prod with `only: master` and `only: prod` respectively.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -1,11 +1,16 @@
-netlify:
-  type: deploy
+image: python:3-slim-stretch
+stages:
+  - deploy
+
+build and push:
```
Comment (pmac) [inline .gitlab-ci.yml]: this job definitely will need a `tags` section and include tags that will match one of our runners.

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -1,11 +1,16 @@
...
+  before_script:
+     - apt-get update -qq && apt-get install -y -qq wget curl
   script:
-    # Stage
-    - curl -X POST -d {} ${WEBHOOK_STAGE}
-    # Prod
-    - curl -X POST -d {} ${WEBHOOK_PROD}
-    # Dead Man's Snitch Ping
-    - curl ${DMS}
-  only:
```
Comment (pmac) [inline .gitlab-ci.yml]: Also without an `only` section this will deploy stage and prod from changes to any branch.

**Pull request closed without merging**